EnableJupyterDisplay.swift

```diff
@@ -0,0 +1,178 @@
+                let jsonString = String(data: jsonData, encoding: .utf8)!
+                return jsonString
+            }
+            catch {
+                return "{}"
+            }
+        }
+    }
+
+    struct Message {
+        var msgType: String = "display_data"
+        var delimiter = "<IDS|MSG>"
+        var key: String = ""
+        var header: Header
+        var metaData: String = "{}"
+        var content: String = "{}"
+        var hmacSignature: String {
+            #if canImport(Cryptor)
+            let data = Data((header.toJSON()+parentHeader+metaData+content).utf8)
+            let keyData = Data(key.utf8)
+            let dataHex = data.map{ String(format: "%02x", $0) }.joined()
+            let keyHex = keyData.map{ String(format: "%02x", $0) }.joined()
+            let hmacKey = CryptoUtils.byteArray(fromHex: keyHex)
+            let hmacData: [UInt8] = CryptoUtils.byteArray(fromHex: dataHex)
+            let hmac = HMAC(using: HMAC.Algorithm.sha256, key: hmacKey).update(byteArray: hmacData)?.final()
+            return CryptoUtils.hexString(from: hmac!)
+            #endif
+            return ""
+        }
+        var messageParts: [KernelCommunicator.BytesReference] {
+            return [
+                bytes(msgType),
+                bytes(delimiter),
+                bytes(hmacSignature),
+                bytes(header.toJSON()),
+                bytes(parentHeader),
+                bytes(metaData),
+                bytes(content)
+            ]
+        }
+        init(content: String = "{}") {
+            header = Header(
+                username: JupyterKernel.communicator.jupyterSession.username,
+                session: JupyterKernel.communicator.jupyterSession.id
+            )
+            self.content = content
+            key = JupyterKernel.communicator.jupyterSession.key
+            #if !canImport(Cryptor)
+            if (!key.isEmpty) {
+                fatalError("""
+                            Unable to import Cryptor to perform message signing.
+                            Add Cryptor as a dependency, or disable message signing in Jupyter as follows:
+                            jupyter notebook --Session.key='b\"\"'\n
+                            """
+                )
+            }
+            #endif
+        }
+    }
+
+    struct PNGImageData: Codable {
+        let image: String
+        let text = "<IPython.core.display.Image object>"
+        init(base64EncodedPNG: String) {
+            image = base64EncodedPNG
+        }
+        private enum CodingKeys: String, CodingKey {
+            case image = "image/png"
+            case text  = "text/plain"
+        }
+    }
+
+    struct MessageContent: Codable {
+        let metadata = "{}"
+        let transient = "{}"
+        let data: PNGImageData
+        init(base64EncodedPNG: String) {
+            data = PNGImageData(base64EncodedPNG: base64EncodedPNG)
+        }
+        public func toJSON() -> String {
```

Comment:
It is not quite common to name Swift methods or properties `toXXX`.  Instead, I would suggest using a computed property called `json`.
```suggestion
        public var json: String {
```